rfc100-resource-lifecycle.md

```diff
@@ -89,6 +89,64 @@ Factors that influence and inform the decision to deprecate a resource include:
 * Resource is added to core chef and documentation is updated.
 * Add deprecated cookbook warnings for conflicting cookbooks.
 
+#### Timeline of resource migration
+
+While we strive to ensure the migration of a resource from a cookbook to core is
+100% compatible, small issues can still arise. In order to ensure that Chef feature
+releases do not cause destabilization, we follow a multi-part timeline for the
+migration. Starting from the top:
+
+1. A resource is added in a cookbook.
+2. That resource is nominated for core inclusion (see above section).
+3. The resource is added to core with an annotation of `defer_to_cookbook_until '>= x.0'`,
+   where `x.0` is the next major version for the cookbook in which the resource
+   will be removed.
+4. Chef releases a minor (features-only) version with the new resource. It will
+   remain inert if the original cookbook is active until the given version.
+5. At some point in the future, the cookbook does the major version release. Users
+   can upgrade to it when they feel ready, otherwise they will see no change in
+   behavior even with the new Chef release.
+6. The following April, as Chef prepares for the yearly major release, all pending
+   `defer_to_cookbook_until` annotations will be removed.
```

Comment:
the default behavior of core resources though is to lose to cookbooks.

which means that cookbooks can override the file resource, which is probably dangerous nonsense, but we've been favoring the case of cookbooks iterating faster and people potentially bugfixing their old chef client versions with new cookbooks.

Reply:
I don't think we'd import a resource into core that overwrote the file resource :)

Reply:
If there was something insane like that I would imagine you would re-write it to more sensible.

Reply:
@lamont-granquist Fair, so we would switch to a new annotation of some kind?

Reply:
Hmm, this might be the reason to switch the annotations around to be driven from the cookbooks side. Need to think through the process >_<